Final code clean-up and minor addition

deswa7310 opened the pull request:
- Added plenty more comments and built upon Tristan's ideas/edits. 
 - Idea of killing the program once the robot detected the finish flag was really good, and I added a delay so it stops once it reaches the flag (not before it). 
 - "containsBlack()" method got scarily close to detecting the black pixels off the side of the image, so I made it scan row 75 instead of 50 just as a precaution (walls might be closer to edge of image in marker's challenge maze). 
 - Also added a small "Finished :)" message before program is killed. 
 - Edited some booleans to make them tidier/more efficient. 
 - Added a semi colon count at the top of main (our program looks massive but main actually only has around 40 semicolons, it's mainly just whitespace and if statements haha). 
 - Edited some comments to make them more accurate (some were outdated).

deswa7310 committed:
Final code clean-up and minor addition

 - Added plenty more comments and built upon Tristan's ideas/edits. 
 - Idea of killing the program once the robot detected the finish flag was really good, and I added a delay so it stops once it reaches the flag (not before it). 
 - "containsBlack()" method got scarily close to detecting the black pixels off the side of the image, so I made it scan row 75 instead of 50 just as a precaution (walls might be closer to edge of image in marker's challenge maze). 
 - Also added a small "Finished :)" message before program is killed. 
 - Edited some booleans to make them tidier/more efficient. 
 - Added a semi colon count at the top of main (our program looks massive but main actually only has around 40 semicolons, it's mainly just whitespace and if statements haha). 
 - Edited some comments to make them more accurate (some were outdated).

Kaolinnite commented:
I'll have a look at this later tonight

Kaolinnite committed:
Update robot.cpp

Changes:
- Edited the documentation lines
- Moved some lines elsewhere; renamed some functions
- Removed the delay function; robot over shot once it reached flag. Change to detection position seemed to be sufficient for stopping the robot
- Edited the message slightly

I can't really think of any other adjustments other than creating a class or struct, but that may require a complete overhaul of the architecture, but as always feel free to submit your own changes

Kaolinnite reviewed (changes requested):
I have created a new pull request with my changes. Aside from this, I don't have any other ideas, at least until someone else makes a change

deswa7310 committed:
Merge pull request #33 from deswa7310/Kaolinnite-patch-2

Update robot.cpp

deswa7310 committed:
Update robot.cpp

Added some more minor documentation edits.
- Updated semicolon count.
- Made vGo static too.
- Updated some comments.
- Note: methods and functions are slightly different. Methods act on the data within objects passed to them (Java only uses methods) whereas functions just act on the data passed to them. Since we aren't using objects or classes, we are only using functions.

Jayman874 committed:
Merge pull request #34 from deswa7310/deswa7310-patch-2

Update robot.cpp

Kaolinnite commented:
All major changes from previous branches approved and applied. Merging now
